### ISSUE TITLE
ERROR: The current node list is empty after upgrading laracast to Intertia

### ISSUE THREAD

**Issue by kossa** [carlosflorencio/laracasts-downloader#123]
Currently laracasts-downloader is not working after the latest upgrade of Laracast to intertia : https://laracasts.com/blog/the-best-of-both-worlds

Any update for this ?

**Comment by Lisendra** [comment]
I can confirm, I'm getting the same error:

```====================================
Authenticating
====================================
ERROR: The current node list is empty.`

**Comment by MicroSystems** [comment]
I got the same issue  ERROR: The current node list is empty

**Comment by willvincent** [comment]
So the problem, obviously, is that the code as it exists today depends on all the data existing in the dom. Which it doesn't now that Laracasts is using inertia. 

The necessary data does, however, appear to be available either in the `data-page` attribute of the `#app` div, or via an xhr request made to the same route, that returns JSON. 

tl;dr - Should be fixable.

**Comment by emilas44** [comment]
Same problem me too....

**Comment by MicroSystems** [comment]
@willvincent any progress fixing this issue ?

**Comment by willvincent** [comment]
@MicroSystems I haven't had any time to try.

**Comment by paolog22** [comment]
up

**Comment by hardikdangar** [comment]
is this fixed ?

**Comment by emilas44** [comment]
Unfortunately not. @willvincent is probably very busy....
Maybe he could try to explain what do we have to do? Maybe there is someone else who could fix this!?

**Comment by paolog22** [comment]
Upon investigating. saw that csrfToken is not in the dom and is now located in js which is the app is still looking at the dom. :), this is for the authentication only. already fix on my end but i am still trying with getting the series

**Comment by AmirRezaM75** [comment]
i'm already working to fix this, but having problem with authentication,any idea how should I log in?
Non of these approaches works :(

```php
$response = $this->client->post(LARACASTS_POST_LOGIN_PATH, [
    'cookies' => $this->cookie,
    'body'    => [
        'email'    => $email,
        'password' => $password,
        '_token' => $csrfToken,
        'remember' => 1,
    ],
    'verify' => false
]);
```

```php
$response = $this->client->post(LARACASTS_POST_LOGIN_PATH, [
    'cookies' => $this->cookie,
    'headers' => [
        'X-CSRF-TOKEN'   => $csrfToken,
    ],
    'body'    => [
        'email'    => $email,
        'password' => $password,
        'remember' => 1,
    ],
    'verify' => false
]);
```

**Comment by willvincent** [comment]
Not sure why things keep getting addressed to me. All I've done is minor tweaks to this code. 🤣

**Comment by carlosflorencio** [comment]
@AmirRezaM75 , the login seems to have changed and the request now sends the payload in JSON format.

It's required to send the Request Header `content-type: application/json` and the body as `{"email":"someemail@gmail.com","password":"efaewr34q5"}`.
